### PR TITLE
Added Logarithmic Halo Potential to externalGravity.cpp + .param file for orbit integration in this potential

### PR DESCRIPTION
**Pull request by jnashworth**

### CODE REVIEW

File: externalGravity.cpp [hunk 1]
```diff
@@ -42,6 +44,10 @@ void ExternalGravity::AddParams(PRM prm)
     dJ6 = 0.0;
     prmAddParam(prm,"dJ6",paramDouble,&dJ6,
                 sizeof(double),"dJ6","<Oblateness coefficient J6 of central body> = 0.0");
+                
+    bLogarithmicHalo = 0;
```
Comment (trquinn) [inline externalGravity.cpp:48]: The compile fails here; I think you forgot to edit externalGravity.h to define dLogarithmicHalo

File: externalGravity.cpp [hunk 2]
```diff
@@ -173,6 +179,35 @@ Vector3D<double> ExternalGravity::applyPotential(GravityParticle *p) const
...
+        double qy = 0.9;
+        double qz = 0.7;
+        double core = sqrt(0.1);
+        p->potential += v0*v0*log(px*px + (py/qy)*(py/qy) + (pz/qz)*(pz/qz) + core*core); // Written as double the true value to account for divide-by-two issue in ChaNGa integration
+        
+        // Acceleration in Cartesian coordinates
+        double ax = -v0*v0*px/(px*px + (py/qy)*(py/qy) + (pz/qz)*(pz/qz) + core*core);
+        double ay = -v0*v0*py/((qy*qy)*(px*px + (py/qy)*(py/qy) + (pz/qz)*(pz/qz) + core*core));
+        double az = -v0*v0*pz/((qz*qz)*(px*px + (py/qy)*(py/qy) + (pz/qz)*(pz/qz) + core*core));
+        Vector3D<double> a = ax*xhat + ay*yhat + az*zhat;
```
Comment (trquinn) [inline externalGravity.cpp:204]: I recommend `Vector3D<double> a(ax, ay, az);` to be a little more straightforward.

File: loghalo_orbit_int.param [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#
```
Comment (trquinn) [inline loghalo_orbit_int.param:1]: I recommend not including this file.

**Review by trquinn — CHANGES_REQUESTED** [review]
See code comments.

File: externalGravity.cpp [hunk 4]
```diff
@@ -173,6 +179,35 @@ Vector3D<double> ExternalGravity::applyPotential(GravityParticle *p) const
...
+        double qy = 0.9;
+        double qz = 0.7;
+        double core = sqrt(0.1);
+        p->potential += v0*v0*log(px*px + (py/qy)*(py/qy) + (pz/qz)*(pz/qz) + core*core); // Written as double the true value to account for divide-by-two issue in ChaNGa integration
+        
+        // Acceleration in Cartesian coordinates
+        double ax = -v0*v0*px/(px*px + (py/qy)*(py/qy) + (pz/qz)*(pz/qz) + core*core);
+        double ay = -v0*v0*py/((qy*qy)*(px*px + (py/qy)*(py/qy) + (pz/qz)*(pz/qz) + core*core));
+        double az = -v0*v0*pz/((qz*qz)*(px*px + (py/qy)*(py/qy) + (pz/qz)*(pz/qz) + core*core));
+        Vector3D<double> a = a(ax, ay, az);
```
Comment (trquinn) [inline externalGravity.cpp:204]: A typo here: This should be `Vector3D<double> a(ax, ay, az);`

File: externalGravity.cpp [hunk 5]
```diff
@@ -173,6 +179,35 @@ Vector3D<double> ExternalGravity::applyPotential(GravityParticle *p) const
         if(idt2 > p->dtGrav)
             p->dtGrav = idt2;
         }
+        
+    if(bLogarithmicHalo) {
+        double px = p->position.x;
+        double py = p->position.y;
+        double pz = p->position.z;
+        double r = p->position.length();
+        // Unit vectors in Cartesian coordinates
```
Comment (trquinn) [inline externalGravity.cpp:188]: These four lines are not needed any more.

**Review by trquinn — CHANGES_REQUESTED** [review]

File: externalGravity.cpp (hunk 5 above)
Comment (jnashworth) [reply externalGravity.cpp]: I assume you mean lines 188-191 (with the xhat, yhat, and zhat vectors); I think I still need lines 185-187.

**Comment by jnashworth** [comment]
Thanks for all the feedback!

**Review by trquinn — APPROVED** [review]

**Pull request merged**